src/middlewared/middlewared/plugins/truecommand/portal.py

```diff
@@ -0,0 +1,165 @@
+import asyncio
+
+from middlewared.service import CallError, job, private, Service
+from middlewared.utils import filter_list
+
+from .connection import TruecommandAPIMixin
+from .enums import PortalResponseState, Status
+
+
+class TruecommandService(Service, TruecommandAPIMixin):
+
+    POLLING_GAP_MINUTES = 5
+
+    @private
+    @job(lock='poll_ix_portal_api_truecommand')
+    async def poll_api_for_status(self, job):
+        await self.middleware.call('truecommand.set_status', Status.CONNECTING.value)
+        config = await self.middleware.call('datastore.config', 'system.truecommand')
+
+        while config['enabled']:
+            try:
+                status = await self.middleware.call('truecommand.poll_once', config)
+            except Exception as e:
+                status = {
```

Comment:
Too broad exception clause for method that can be aborted: you'll ignore CancelledError this way